### PR TITLE
feat(jupyter-datascience-notebooks): add a new template jupyter-datascience-notebooks

### LINKED ISSUE #2: Add jupyter-datascience-notebooks template
This seems to be a good first template to add. It's been discussed in https://github.com/orgs/devcontainers-community/discussions/1#discussioncomment-5351589

Here's a tentative todo thingy:
- [x] Reach out to the original maintainer @nathancarter
- [x] Get their permission to move/improve/do stuff
- [x] Add the actual code content into this repo
- [ ] OPTIONAL: Deprecate and/or document the change in the original location? unsure of how to go about this. I wasn't involved with devcontainers-related stuff until Nov 2022 which was _after_ the deprecation of https://github.com/microsoft/vscode-dev-containers i think... @eitsupi will probably need to assist me and/or do this

@nathancarter Are you interested in being added as a collaborator, or would you prefer to make it more of a "transferring the torch" kinda thing?

### PR DESCRIPTION
**Pull request by eitsupi**
Closes #2

Add a template for the purpose of replacing the legacy template that is in https://github.com/microsoft/vscode-dev-containers/tree/main/containers/jupyter-datascience-notebooks.

And update some Lint settings of CI.

### CODE REVIEW

File: src/jupyter-datascience-notebooks/NOTES.md
Comment (jcbhmr) [inline src/jupyter-datascience-notebooks/NOTES.md:1]: I'd love to see more direct documentation! 🤩

Something like: (ignore my link formatting; it doesn't matter)

```md
<!-- markdownlint-disable MD041 -->

This template was originally created by [@nathancarter] and other contributors
in the [microsoft/vscode-dev-containers] repository. It has since been landed
here. 🌠

## Choosing a base image

You can choose from many base images to tailor the container to your project's
needs. If you don't know what this means, that's OK! Just choose the
`datascience-notebook` option. It has libraries for data analysis from the
Julia, Python, and R communities. 🚀

📚 You can learn more about what each of these base images is and what features
each of them in [Selecting an Image — Docker Stacks documentation]

<!-- prettier-ignore-start -->
[@nathancarter]: https://github.com/nathancarter
[microsoft/vscode-dev-containers]: https://github.com/microsoft/vscode-dev-containers
[Selecting an Image — Docker Stacks documentation]: https://jupyter-docker-stacks.readthedocs.io/en/latest/using/selecting.html
<!-- prettier-ignore-end -->
```

File: test/jupyter-datascience-notebooks/test.sh
Comment (jcbhmr) [inline test/jupyter-datascience-notebooks/test.sh:2]: Out of curiosity, why is this here? 🤔 consistency across different dirs?

**Review by jcbhmr — APPROVED** [review]
🥇 Looks good enough to merge! 👍 More stuff can be added in the future (like better docs, authors/maintainers names, better comments, etc.). I think this is a good first template.

I know you mentioned something about "what information should go in the NOTES.md" somewhere, so you might be interested in https://github.com/devcontainers-community/templates/discussions/8

File: test/jupyter-datascience-notebooks/test.sh
Comment (eitsupi) [reply test/jupyter-datascience-notebooks/test.sh:2]: I think this script expects to call `test-utils.sh`, so it needs to go to a specific directory.

File: src/jupyter-datascience-notebooks/NOTES.md
Comment (eitsupi) [reply src/jupyter-datascience-notebooks/NOTES.md:1]: Thanks for taking a look at this. Your document looks great and I would like to adopt it.

**Comment by eitsupi** [comment]
Thanks for the review.
It would be great if we could include the scope of adding Template in the README before merging this. (I am in no hurry to merge this)

**Comment by jcbhmr** [comment]
> (I am in no hurry to merge this)

Good attitude to take. 🎖 I appreciate your counterweight to my zeal 💕

File: src/jupyter-datascience-notebooks/.devcontainer/devcontainer.json
Comment (eitsupi) [inline src/jupyter-datascience-notebooks/.devcontainer/devcontainer.json:28]: These settings are probably worth mentioning in the documentation, as they may not be necessary when using VS Code (I've just copied them for now)

**Comment by eitsupi** [comment]
It has become apparent that some configuration changes are needed in linter CI.
I will do that in another PR.

File: src/jupyter-datascience-notebooks/devcontainer-template.json
Comment (eitsupi) [inline src/jupyter-datascience-notebooks/devcontainer-template.json:39]: I have a feeling these are not platforms, but are copied straight from the original template.......

**Comment by eitsupi** [comment]
@jcbhmr I think I have finished the work.

If you look at it and it looks like there are no major problems, let's merge it and submit a PR to the index <https://github.com/devcontainers/devcontainers.github.io>.

File: src/jupyter-datascience-notebooks/devcontainer-template.json
Comment (jcbhmr) [reply src/jupyter-datascience-notebooks/devcontainer-template.json:39]: opened https://github.com/devcontainers/templates/issues/145 because I remembered you saying something like "the devcontainers official people get paid to do this" so I figured might as well ask, right? cant hurt.

Comment (eitsupi) [reply src/jupyter-datascience-notebooks/devcontainer-template.json:39]: Thanks for opening that!
I too don't think this field is currently being used for anything.

So we can keep this as it is for now.

**Comment by eitsupi** [comment]
@jcbhmr CI worked and published successfully, should I send the PR to the index?

**Comment by eitsupi** [comment]
@jcbhmr Any thoughts?

**Comment by jcbhmr** [comment]
Sounds great! 👍

**Comment by eitsupi** [comment]
devcontainers/devcontainers.github.io#202

**Pull request merged**
